### DISCUSSION TITLE
Can jupedsim monitor more pedestrian information in a specific area?

### DISCUSSION THREAD

**Discussion by verse-y** [PedestrianDynamics/jupedsim#1399] [General]
I saw in the documentation that Jupedsim can measure N-t data with measurement_line and calculate the speed and density change for each frame with PedPy. So can Jupedsim monitor more pedestrian information at a point or a circular area? Such as the direction of pedestrian movement, the size of pedestrian speed, and the number of pedestrians. If such a function is available, evacuation guidance based on Wireless Sensor Network (WSN) assistance may be realized. And is there a concept of capacity in the model for a particular area (room, corridor, or other space facility)? When the capacity of this area exceeds its limit, pedestrians can be guided to other regions, which will be of great practical significance and help to study the optimization of evacuation paths.

**Comment by Ozaq** [comment]
Two things: 

1.) JuPedSim has no functionality to compute n/t curve, in fact JuPedSim has no analysis capabilities. This functionality is from a python module called PedPy. See https://pypi.org/project/PedPy/

2.) You can do what you describe with JuPedSim and some python code. JuPedSim allows you to query all agents in a polygon or circle or simply to iterate over all agents. 
Each agents behavior can be modified whenever you want. You can create new journeys or stages at any point in time.
